### PR TITLE
Add CocoaPods spec file to simplify project setup.

### PR DESCRIPTION
**Pull request by christoph-jerolimov**
For example this fix the setup requirement that the Xcode header search path must be changed anymore / and anytime the node_modules folder was resetted.

See changed ios/installation for more details.

### CODE REVIEW

**Comment by bsudekum** [comment]
This looks good. Going to test this weekend.

**Comment by christoph-jerolimov** [comment]
FYI, In my project I added the Podfile within the `ios` folder with a **clean** Xcode project and CocoaPods.

My `AppDelegate`:

``` objc
#import <React/RCTRootView.h>

...
    NSURL *mainBundle = [[NSBundle mainBundle] URLForResource:@"main" withExtension:@"jsbundle"];

    // Just override the bundle URL, use debug env later,
    mainBundle = [NSURL URLWithString:@"http://localhost:8081/index.ios.bundle?platform=ios&dev=true"];

    RCTRootView *rootView = [[RCTRootView alloc] initWithBundleURL:mainBundle
                                                        moduleName:@"AppName"
                                                 initialProperties:nil
                                                     launchOptions:launchOptions];

    self.window = [[UIWindow alloc] initWithFrame:[UIScreen mainScreen].bounds];
    UIViewController *rootViewController = [[UIViewController alloc] init];
    rootViewController.view = rootView;
    self.window.rootViewController = rootViewController;
    [self.window makeKeyAndVisible];
```

My `Podfile`:

``` ruby
source 'https://github.com/CocoaPods/Specs.git'

pod 'React', :path => '../node_modules/react-native'
pod 'React/RCTGeolocation', :path => '../node_modules/react-native'
pod 'React/RCTImage', :path => '../node_modules/react-native'
pod 'React/RCTNetwork', :path => '../node_modules/react-native'
pod 'React/RCTText', :path => '../node_modules/react-native'
pod 'React/RCTWebSocket', :path => '../node_modules/react-native'

#pod 'RCTMapboxGL', :path => '../node_modules/react-native-mapbox-gl/ios' # npm version
pod 'RCTMapboxGL', :path => '../../react-native-mapbox-gl/ios' # local clone
```

Maybe I will create a blog posts about this later.

**Comment by bsudekum** [comment]
I just released 1.2.0. I'm going to save this for the next release to better triage bug reports.

**Comment by nevir** [comment]
Awesome, thanks for the cocoapods setup, it'll make it much easier to consume this library!

**Comment by chris-erickson** [comment]
That setup phase was rough, can't wait for this! :chocolate_bar:

**Comment by edvinasbartkus** [comment]
:+1: awesome!

**Comment by chris-erickson** [comment]
@bsudekum When can we expect the next release?

**Comment by bsudekum** [comment]
@jerolimov how will this effect someone who has an existing project and they'll adding this module? Will they have to use the `project.xcworkspace` file now? I'm not well versed in cocoapod workflows. 

/cc @incanus

**Comment by christoph-jerolimov** [comment]
No, this effect the users only if they explicit import the spec file in their Podfile. So there is no risk to merge it. If you want use the "react-native / old way" and modify the project yourself it will work.

With the spec file you have the option to use Cocoapods. CocoaPods will create a new Xcode project called Pods which is generated and just import this Pods project into your Project. This makes updates really easy.

Both works parallel without trouble and any user can choice what they use in their project.

(I'm only unsure if we should move and rename the file into the top folder, `react-native-mapbox-gl/react-native-mapbox-gl.podspec` instead of `react-native-mapbox-gl/ios/RCTMapboxGL.podspec`.)

Should I create a demo project?

**Comment by jerolimov** [comment]
@bsudekum Maybe you can talk to Justin R. Miller and Minh Nguyễn (didn't know their username here) which maintain the `Mapbox-iOS-SDK` project on cocoapods: https://cocoapods.org/pods/Mapbox-iOS-SDK

**Comment by christoph-jerolimov** [comment]
@bsudekum Maybe you can talk to @incanus and @1ec5 which maintain already the `Mapbox-iOS-SDK` project on cocoapods: https://cocoapods.org/pods/Mapbox-iOS-SDK

File: ios/install.md [hunk 1]
```diff
@@ -1,8 +1,21 @@
...
+
 * `react-native ^v0.4.3` is required
 
+## Use CocoaPods
+
+If you already use [CocoaPods](https://cocoapods.org/) in your react-native project you add the native
+part of Mapbox GL with just one line of code:
+
+1. `npm install react-native-mapbox-gl --save`
+1. Add `pod 'RCTMapboxGL', :path => 'node_modules/react-native-mapbox-gl/ios'` to your `Podspec` file and re-run `pod install`.
```
Comment (1ec5) [inline ios/install.md:13]: By the way, this points to a local, npm-based installation of RCTMapboxGL. I guess it isn’t strictly necessary to install the pod from cocoapods.org.

**Comment by 1ec5** [comment]
@bsudekum, as @jerolimov says, there’s no impact for someone who wants to follow the existing installation workflow. This would be more convenient for someone who’s already using CocoaPods for other modules. I haven’t tried the pod locally, but the Podfile looks good to me.

File: ios/install.md (hunk 1 above)
Comment (christoph-jerolimov) [inline ios/install.md:13]: Your right, publishing it on CocoaPods is not required and maybe redundant.

The release notes of react-native 0.13 contains some infos how facebook will handle this. And my own tries shows the same problems.

https://github.com/facebook/react-native/releases/tag/v0.13.0

**Comment by christoph-jerolimov** [comment]
Thanks @bsudekum :+1:

**Pull request merged**